src/nccl_ofi_scheduler.c

```diff
@@ -22,6 +22,43 @@ static inline size_t sizeof_schedule(int num_rails)
 		+ num_rails * sizeof(nccl_net_ofi_xfer_info_t);
 }
 
+/*
+ * @brief  This function calculates the optimal number of stripes
+ * for the payload size based on the min_stripe_size.
+ * It first determines the initial number of stripes based on the message size
+ * and the minimum stripe size, ensuring that the result is at least 1 and at
+ * most the number of available rails.
+ * The function then adjusts the number of stripes to be the largest factor of
+ * the number of rails that is less than or equal to the initial number of stripes.
+ *
+ * @param	size
+ * 		The size of the message being transmitted.
+ * @param 	scheduler_p
+ * 		Pointer to threshold scheduler
+ * @param 	num_rails
+ * 		The number of available rails for transmission.
+ *
+ * @return	Returns the adjusted number of stripes.
+ *
+ */
+static int get_num_stripes(nccl_net_ofi_threshold_scheduler_t *scheduler_p, size_t size, int num_rails)
+{
+	/* Number of stripes is atleast 1 for zero-sized messages and at most equal to num of rails */
+	int num_stripes =
+		(int)NCCL_OFI_MAX(1, NCCL_OFI_MIN(NCCL_OFI_DIV_CEIL(size, scheduler_p->min_stripe_size), (unsigned)num_rails));
+
+	/* Start the loop from num_stripes and skip 1, as num_rails % 1 is always true.
+	 * This avoids the overhead of the mod operation in latency-sensitive cases.
+	 */
+	for (int i = num_stripes; i >= 1; i--) {
```

Comment:
As I said before, this should really be `i > 1`, not `i >= 1`.

Reply:
Sorry. I have corrected now.